### PR TITLE
fix: owlbot-cli image sha to be locked (hermetic)

### PR DESCRIPTION
**Pull request by diegomarquezp**
owlbot-cli currently [defaults](https://github.com/googleapis/sdk-platform-java/blob/b861f19a824353da8f585151924813d9197b3c9b/library_generation/postprocess_library.sh#L84) to `latest`, which is not hermetic. This PR proposes using a configuration file to lock the image to a specific SHA (the SHA cannot be inferred from configuration files in target repositories).

As of November 2023, the copy-code command implementation has had [6 commits](https://github.com/googleapis/repo-automation-bots/commits/main/packages/owl-bot/src/copy-code.ts) this year.

### CODE REVIEW

File: library_generation/configuration/owlbot-cli-sha [hunk 1]
```diff
@@ -0,0 +1 @@
+sha256:623647ee79ac605858d09e60c1382a716c125fb776f69301b72de1cd35d49409
```
Comment (JoeWang1127) [inline library_generation/configuration/owlbot-cli-sha:1]: Can you add a comment about the date of this image?

**Review by JoeWang1127 — APPROVED** [review]

File: library_generation/configuration/owlbot-cli-sha (hunk 1 above)
Comment (diegomarquezp) [reply library_generation/configuration/owlbot-cli-sha:2]: Done

Comment (blakeli0) [reply library_generation/configuration/owlbot-cli-sha:2]: How do we plan to update this? Manually or automatically similar to owlbot post-processing image?

Comment (diegomarquezp) [reply library_generation/configuration/owlbot-cli-sha:2]: It's setup for manual updates. We would have to modify this file. We can also have it as a custom argument to generate_library.sh that would replace the default one found here.

**Review by blakeli0 — APPROVED** [review]
LGTM. Let's follow up with a separate PR that use renovate-bot to automatically update the sha.

**Pull request merged**